update trouble api

jamessizeland opened the pull request:
waiting on https://github.com/lulf/microbit-bsp/pull/19 to merge

jamessizeland committed:
update trouble api

jamessizeland committed:
update dependencies

lure23 committed:
suggestion on some deps cleanup (#7)

Co-authored-by: Asko Kauppi <akauppi@gmail.com>

jamessizeland committed:
update to new trouble api structure

jamessizeland commented:
I'm seeing weird behaviour on this update, the server keeps timing out on connection.  @lure23 @lulf do you get a similar behaviour on your microbits? I can't see what is different in this to running the trouble example in microbit-bsp (which does work).  So I'm wondering if it's indicating a bug somewhere...

```
39.388824 INFO  [host] disconnection event on handle 2, reason: Connection Timeout
└─ trouble_host::host::{impl#4}::run_with_handler::{async_fn#0} @ /home/.cargo/git/checkouts/trouble-148cf413cadab653/a1daee2/host/src/fmt.rs:143 
39.389007 INFO  Gatt server task finished
```

lure23 commented:
> I'm seeing weird behaviour on this update

Sorry, but I don't have a microbit to test with.

lulf commented:
> I'm seeing weird behaviour on this update, the server keeps timing out on connection. @lure23 @lulf do you get a similar behaviour on your microbits? I can't see what is different in this to running the trouble example in microbit-bsp (which does work). So I'm wondering if it's indicating a bug somewhere...
> 
> ```
> 39.388824 INFO  [host] disconnection event on handle 2, reason: Connection Timeout
> └─ trouble_host::host::{impl#4}::run_with_handler::{async_fn#0} @ /home/.cargo/git/checkouts/trouble-148cf413cadab653/a1daee2/host/src/fmt.rs:143 
> 39.389007 INFO  Gatt server task finished
> ```

@jamessizeland Could you test with https://github.com/embassy-rs/trouble/pull/270 ? I'm wondering if it's a symptom that the outbound reply is dropped due to outbound buffers or something (I've not tried to debug this, so it's just guessing). If there is a lot of events and you want to make sure they all get replied to, you need to explicitly call `event.accept()?.send().await` so that it will wait until the reply is put into the outbound queue.

jamessizeland commented:
I'll do so later yep ty!

jamessizeland commented:
I switched to your branch and it connected and displayed characteristics in nrf-connect properly. But then I switched back to the rev that wasn't working yesterday and it connects just fine, so I'm unsure what the issue was!

jamessizeland committed:
add patch
